City-of-Helsinki/helsinki-paatokset
# PP-0: File proxy for Ahjo documents

**To test**
- Checkout branch, run `make shell` and then `drush cr; drush cim -y`
- Import some meeting documents:
  - `drush mim ahjo_meetings:latest --update` (run `drush ap:fs -v` if you haven't stored the static files yet
  -  The `--update` flag is important for refreshing the local URLs for the documents
- Connect to the VPN and authenticate with the Ahjo API: https://helsinki-paatokset.docker.so/fi/admin/ahjo-open-id (or just refresh the token if you've already authenticated)
  - We can test this without the VPN on dev and test 
- Find some ahjo documents and edit them: https://helsinki-paatokset.docker.so/fi/admin/content/media?name=&type=ahjo_document&status=All&langcode=All
  - The  file URIs should be in this format: https://helsinki-paatokset.docker.so/fi/ahjo-proxy/document/[ID]
  - Copy the URL and open it in a browser. The PDF file should either open or get downloaded
  - Try logging out of Drupal and opening the URL. You should get an access denied error
